cpp/src/arrow/ipc/writer.h

```diff
@@ -75,6 +76,19 @@ struct WriteStats {
   /// Number of replaced dictionaries (i.e. where a dictionary batch replaces
   /// an existing dictionary with an unrelated new dictionary).
   int64_t num_replaced_dictionaries = 0;
+
+  /// initial and serialized (may be compressed) body lengths for record batches
+  /// these values show the total sizes of all record batch body lengths
+  int64_t raw_body_length = 0;
+  int64_t serialized_body_length = 0;
```

Comment:
In the Arrow C++ APIs, "length" generally points to the logical number of elements (see e.g. `Array::length()`), while "size" points to the physical size in bytes (as in `Buffer::size()`). So I think this should be:
```c++
int64_t total_raw_body_size = 0;
int64_t total_compressed_body_size = 0;
```

Reply:
Sometimes it is not compressed, only serialized. Should we name it as total_serialized_body_size?

Reply:
Yes, why not.